worker/src/handlers/schedule.ts

```diff
@@ -1,181 +1,171 @@
-  const core_analytics: Record<string, any> = {};
-  const storedAnalytics =
-    (await KV.get<{ [key: string]: CurrentAnalytics }>(
-      "core_analytics",
-      "json"
-    )) || {};
-  const storedData = await listStoredData();
-  console.log(JSON.stringify(storedData));
-  //const currentDataset: CurrentAnalytics = generateCurrentDataset(storedData);
-
-  const currentDate = formatDate(new Date());
-  const currentDateObj = new Date(
-    currentDate.year,
-    currentDate.month,
-    currentDate.day,
-    currentDate.hour
-  );
-  const timestampString = String(currentDateObj.getTime());
-  //const storeKey = `history:${timestampString}`;
+  await processQueue();
+}
 
-  const dataKeys = Object.keys(storedAnalytics);
-  const lastDataEntry = storedAnalytics[dataKeys[dataKeys.length - 1]];
+async function processQueue(): Promise<void> {
+  const queue = (await KV.get<Queue>("queue", "json")) || {
+    entries: [],
+    data: baseQueueData,
+  };
 
-  for (const key of dataKeys) {
-    core_analytics[key] = {
-      active_installations: storedAnalytics[key].active_installations,
-      installation_types: storedAnalytics[key].installation_types,
-    };
+  if (!queue.data || Object.keys(queue.data).length === 0) {
+    queue.data = baseQueueData;
   }
 
-  core_analytics[timestampString] = {
-    ...lastDataEntry,
-    active_installations: storedData,
-  };
+  if (queue.entries.length === 0) {
+    // No entries, get list
+    console.log("No entries, get list");
+    queue.entries = await listKV("uuid");
+  }
 
-  //await KV.put(storeKey, JSON.stringify(currentDataset));
-  await KV.put("core_analytics", JSON.stringify(core_analytics));
+  // Prosess the first 850 entries in the array
+  for (const entryKey of queue.entries.splice(0, 850)) {
+    let entryData;
+    try {
+      entryData = await KV.get<SanitizedPayload>(entryKey, "json");
+    } catch (e) {
+      console.log(e);
+    }
+
+    if (entryData !== undefined && entryData !== null) {
+      queue.data = combineEntryData(queue.data, entryData);
+    }
+  }
+
+  if (queue.entries.length === 0) {
+    // No more entries, store and reset queue data
+    console.log("No more entries, store and reset queue data");
+    const core_analytics: Record<string, any> = {};
+    const timestampString = String(new Date().getTime());
+
+    const queue_data = processQueueData(queue.data);
+    const storedAnalytics =
+      (await KV.get<{ [key: string]: CurrentAnalytics }>(
+        "core_analytics",
+        "json"
+      )) || {};
+
+    for (const key of Object.keys(storedAnalytics)) {
+      core_analytics[key] = {
+        active_installations: storedAnalytics[key].active_installations,
+        installation_types: storedAnalytics[key].installation_types,
+      };
+    }
+
+    core_analytics[timestampString] = queue_data;
+
+    await KV.put(`history:${timestampString}`, JSON.stringify(queue_data));
+    await KV.put("core_analytics", JSON.stringify(core_analytics));
```

Comment:
We should add a prefix to all keys that we set.

Reply:
prefix only make sense if you have multiple for the same thing, "core_analytics" does not have anything similar, same with "queue"

prefix both with "operational" ?

Reply:
core: 😅

Reply:
I guess that makes sense. Use a colon if it can be multiple